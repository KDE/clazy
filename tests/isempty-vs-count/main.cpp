#include <QtCore/QList>
#include <QtCore/QMap>
#include <QtCore/QMultiMap>

QList<int> getList() {return {};}

void test()
{
    QList<int> list;
    if (list.count()) {}   // Warning
    if (!list.count()) {}   // Warning
    bool b1 = list.count(); // Warning
    bool b2 = list.size(); // Warning
    if (list.indexOf(1)) {} // OK

    QMap<int, int> map;
    if (map.count()) {} // Warning
    if (map.count(2)) {} // Warning

    QMultiMap<int, int> multiMap;
    if (multiMap.count()) {} // Warning
    if (multiMap.count(2)) {} // Warning
    if (multiMap.count(2, 3)) {} // Warning

    if (getList().count()) {} // Warning
}

