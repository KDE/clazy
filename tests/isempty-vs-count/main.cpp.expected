isempty-vs-count/main.cpp:10:9: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:11:10: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:12:15: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:13:15: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:17:9: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:18:9: warning: use contains() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:21:9: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:22:9: warning: use contains() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:23:9: warning: use contains() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:25:9: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:30:9: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:31:9: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
isempty-vs-count/main.cpp:32:9: warning: use isEmpty() instead [-Wclazy-isempty-vs-count]
